crates/transport-ws/src/native.rs

```diff
@@ -63,8 +72,13 @@ impl PubSubConnect for WsConnect {
     async fn connect(&self) -> TransportResult<alloy_pubsub::ConnectionHandle> {
         let request = self.clone().into_client_request();
         let req = request.map_err(TransportErrorKind::custom)?;
-        let (socket, _) =
-            tokio_tungstenite::connect_async(req).await.map_err(TransportErrorKind::custom)?;
+        let (socket, _) = tokio_tungstenite::connect_async_with_config(
+            req,
+            Some(self.config.unwrap_or_default()),
```

Comment:
In case that changes we should stick to providing an option here
```suggestion
            self.config,
```